online/src/main/scala/ai/chronon/online/CatalystUtil.scala

```diff
@@ -191,11 +191,31 @@ class CatalystUtil(expressions: collection.Seq[(String, String)],
         projectFunc
       }
       case ProjectExec(projectList, childPlan) => {
-        val unsafeProjection = UnsafeProjection.create(projectList, childPlan.output)
-        def projectFunc(row: InternalRow): Option[InternalRow] = {
-          Some(unsafeProjection.apply(row))
+        childPlan match {
+          case whc @ WholeStageCodegenExec(fp @ FilterExec(condition, child)) =>
```

Comment:
I can remove code repetition with the other `WholeStageCodegenExec` branch, but the general idea is here & folks can review.